app/assets/javascripts/index/directions.js

```diff
@@ -109,26 +125,61 @@ OSM.Directions = function (map) {
 
       var viewbox = map.getBounds().toBBoxString(); // <sw lon>,<sw lat>,<ne lon>,<ne lat>
 
-      $.getJSON(OSM.NOMINATIM_URL + "search?q=" + encodeURIComponent(endpoint.value) + "&format=json&viewbox=" + viewbox, function (json) {
-        endpoint.awaitingGeocode = false;
-        endpoint.hasGeocode = true;
-        if (json.length === 0) {
-          input.addClass("is-invalid");
-          alert(I18n.t("javascripts.directions.errors.no_place", { place: endpoint.value }));
-          return;
-        }
+      if (endpoint.value.match(/^([+-]?\d+(\.\d*)?)(?:\s+|\s*[/,]\s*)([+-]?\d+(\.\d*)?)$/)) {
+        var latlng_array = endpoint.value.split(/[/,]/);
+        $.getJSON(OSM.NOMINATIM_URL + "reverse?lat=" + latlng_array[0] + "&lon=" + latlng_array[1] + "&format=json&viewbox=" + viewbox, function (json) {
+          endpoint.awaitingGeocode = false;
+          endpoint.hasGeocode = true;
 
-        endpoint.setLatLng(L.latLng(json[0]));
+          endpoint.setLatLng(L.latLng(latlng_array), false);
 
-        input.val(json[0].display_name);
+          if (!json || !json.display_name) {
+            input.addClass("is-invalid");
+            alert(I18n.t("javascripts.directions.errors.no_place", { place: endpoint.value }));
+            return;
+          }
 
-        geocodeCallback();
-      });
+          if (endpoint.latlng.distanceTo(L.latLng(json.lat, json.lon)) > 200.0) {
+            input.val(endpoint.value);
+          } else {
+            endpoint.value = json.display_name;
+            input.val(json.display_name);
+          }
+
+          geocodeCallback();
+        });
+      } else {
+        $.getJSON(OSM.NOMINATIM_URL + "search?q=" + encodeURIComponent(endpoint.value) + "&format=json&viewbox=" + viewbox, function (json) {
+          endpoint.awaitingGeocode = false;
+          endpoint.hasGeocode = true;
+          if (!json || json.length === 0) {
```

Comment:
This can be shortened as `!json?.length`

Reply:
`let/const`, template literals, `?.` etc - we should decide on these in a separate issue, not in this PR.

Reply:
If it needs a separate issue, we can create it, however I didn't mean refactoring of the old code (which definitely would be a separate issue), but to change them in the new code. Also, if somewhere we have coding style guidelines, please, link it and I'll create an issue linked to it.

Reply:
Looked at what is already in use by included libraries:

- `let/const` - used by Bootstrap and Turbo
- `?.` - used by Turbo
- template literals - used by Bootstrap and Turbo